pulp_python/app/serializers.py

```diff
@@ -232,13 +235,17 @@ def deferred_validate(self, data):
         _data['version'] = metadata.version
         _data['filename'] = filename
         _data['sha256'] = artifact.sha256
+        data["pulp_domain_id"] = artifact.pulp_domain_id
+        data["_pulp_domain_id"] = artifact.pulp_domain_id
 
         data.update(_data)
 
         return data
 
     def retrieve(self, validated_data):
-        content = python_models.PythonPackageContent.objects.filter(sha256=validated_data["sha256"])
+        content = python_models.PythonPackageContent.objects.filter(
+            sha256=validated_data["sha256"], _pulp_domain=get_domain()
```

Comment:
Just curious, my fuzzy understanding of the purpose of `_pulp_domain` is that we need the uniqueness constraint which is only available within a single table (thus - it needs to be on the plugin model) but we also need to be able to access it from pulpcore code, so we duplicate the data.

I think that's correct, please correct me if I'm missing something.

If that's the case, then shouldn't this query (and any others like it) be using `pulp_domain` (from the plugin, not core) so that Django doesn't have to join across tables?